<launch>

    <arg name="env" default="wrench"/>
    <arg name="ref_pcd" default="/home/cyborg/catkin_ws/src/daedalus/arm2d2_description/meshes/wrench.pcd"/>
    <arg name="gui" default="false"/>
    <arg name="output" default="screen"/>

    <rosparam file="$(find apogee_vision)/config/particle_filter_config.yaml" command="load"/>
    <rosparam file="$(find daedalus_core)/config/schedule_rates.yaml" command="load"/>

    <include file="$(find apogee_moveit_config)/launch/moveit_rviz.launch"/>


    <include file="$(find mujoco_ros)/launch/sim.launch">
        <arg name="env" value="$(arg env)"/>
        <arg name="fixed_cam" value="false"/>
    </include>

    <node name="point_cloud" pkg="apogee_vision" type="point_cloud" output="screen"/>

    <node if="$(arg gui)" name="tracker" pkg="apogee_vision" type="object_tracking_gui" output="$(arg output)" args="$(arg ref_pcd)"/>
    <node unless="$(arg gui)" name="tracker" pkg="apogee_vision" type="object_tracking" output="$(arg output)" args="$(arg ref_pcd)"/>
    <node name="object_marker_publisher" pkg="apogee_vision" type="rviz_tracked_object"/>

</launch>